caiman/utils/visualization.py

```diff
@@ -1217,3 +1218,143 @@ def update(val):
     s_cn_min.on_changed(update)
     s_pnr_max.on_changed(update)
     s_pnr_min.on_changed(update)
+
+
+def get_rectangle_coords(im_dims: ArrayLike, 
+                         stride: int, 
+                         overlap: int) -> Tuple[np.ndarray, np.ndarray]:
+    """
+    Extract rectangle (patch) coordinates: a helper function used by view_quilt().
+    
+    Given dimensions of summary image (rows x colums), stride between patches, and overlap
+    between patches, returns row coordinates of the patches in patch_rows, and column 
+    coordinates patches in patch_cols. This is meant to be used by plot_patches().
+       
+    Args:
+        im_dims: array-like
+            dimension of image (num_rows, num_cols)
+        stride: int
+            stride between patches in pixels
+        overlap: int
+            overlap between patches in pixels
+    
+    Returns:
+        patch_rows: ndarray
+            num_patch_rows x 2 array, where row i contains onset and offset row pixels for patch row i
+        patch_cols: ndarray
+            num_patch_cols x 2 array, where row j contains onset and offset column pixels for patch column j
+            
+    Note: 
+        Currently assumes square patches so takes in a single number for stride/overlap. 
+    """
+    patch_width = overlap + stride
+    
+    patch_onset_rows = np.array(list(range(0, im_dims[0] - patch_width, stride)) + [im_dims[0] - patch_width])
+    patch_offset_rows = patch_onset_rows + patch_width
+    patch_offset_rows[patch_offset_rows > im_dims[0]-1] = im_dims[0]-1
+    patch_rows = np.column_stack((patch_onset_rows, patch_offset_rows))
+    
+    patch_onset_cols = np.array(list(range(0, im_dims[1] - patch_width, stride)) + [im_dims[1] - patch_width])
+    patch_offset_cols = patch_onset_cols + patch_width 
+    patch_offset_cols[patch_offset_cols > im_dims[1]-1] = im_dims[1]-1
+    patch_cols = np.column_stack((patch_onset_cols, patch_offset_cols))
+    
+    return patch_rows, patch_cols
+
+
+def rect_draw(row_minmax: ArrayLike, 
+              col_minmax: ArrayLike, 
+              color: Optional[str]='white', 
+              alpha: Optional[float]=0.3, 
+              ax: Optional[Any]=None) -> Tuple[Any, Any]:
+    """
+    Draw a single transluscent rectangle on given axes object.
+    
+    Args:
+        row_minmax: array-like
+            [row_min, row_max] -- 2-elt int bounds for rect rows
+        col_minmax: array-like 
+            [col_min, col_max] -- int bounds for rect cols
+        color : string
+            rectangle color, default 'yellow'
+        alpha : float
+            rectangle alpha (0. to 1., where 1 is opaque), default 0.3
+        ax : pyplot.Axes object
+            axes object upon which rectangle will be drawn, default None
+    
+    Returns:
+        ax (Axes object)
```

Comment:
If we don't want to specify types for these, can we say a little bit more about what we expect them to be? matplotlib.axes?